tests/php82_readonly_class.phpt

```diff
@@ -0,0 +1,27 @@
+--TEST--
+readonly class support in php 8.2+
+--SKIPIF--
+<?php if (!class_exists('AllowDynamicProperties')) die('skip PHP >=8.2 only'); ?>
+--FILE--
+<?php
+require __DIR__ . '/../util.php';
+
+$code = <<<'PHP'
+<?php
+readonly class X {}
+PHP;
+
+$node = ast\parse_code($code, $version=85);
+echo ast_dump($node), "\n";
+--EXPECTF--
+AST_STMT_LIST
+    0: AST_CLASS
+        flags: CLASS_READONLY | CLASS_NO_DYNAMIC_PROPERTIES (%d)
```

Comment:
I don't think we should expose CLASS_NO_DYNAMIC_PROPERTIES. It's an implementation detail (it might make sense to update php-src to only set this flag in the compiler, not the parser).

Reply:
> I don't think we should expose CLASS_NO_DYNAMIC_PROPERTIES. It's an implementation detail (it might make sense to update php-src to only set this flag in the compiler, not the parser).

Done. Originally, no examples of changing flags came to mind, but I guess there's the virtual `NULLABLE_TYPE` nodes and removing the IS_NULLABLE bit flag.

It should be doable in php-src zend_compile.c which runs before opcache, but doesn't really seem like it'll affect extensions in practice, especially with php 9.0 forbidding dynamic properties by default.